src/compiler/types.ts

```diff
@@ -1904,6 +1904,8 @@ namespace ts {
         isolatedSignatureType?: ObjectType; // A manufactured type that just contains the signature for purposes of signature comparison
     }
 
+    export type TypeSystemEntity = Symbol | Type | Signature;

```

Comment:
Actually, this does't even need to be exported or in `types.ts`, just put it where you declare `resolutionTargets`